src/taxpasta/infrastructure/application/malt/malt_profile_reader.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2022, Moritz E. Beber, Maxime Borry, Jianhong Ou, Sofia Stamouli.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Provide a reader for malt profiles."""
+
+import pandas as pd
+from pandera.typing import DataFrame
+
+from taxpasta.application import ProfileReader, ProfileSource
+
+from .malt_profile import MaltProfile
+
+
+class MaltProfileReader(ProfileReader):
+    """Define a reader for MALT-rma2info profiles."""
+
+    LARGE_INTEGER = int(10e6)
+
+    @classmethod
+    def read(cls, profile: ProfileSource) -> DataFrame[MaltProfile]:
+        """Read a MALT-rma2info taxonomic profile from a file."""
+        nb_expected_columns = 2
+        result = pd.read_table(
+            filepath_or_buffer=profile,
+            compression="gzip",
```

Comment:
When the filename ends in `.gz` this is automatically recognized by pandas and decompressed. I'm wondering if that should be added as a parameter to the `read` function so that it can be determined more flexibly.

Reply:
For the rma2info, the output is gzip compressed. I've anyway changed it to let pandas infer what to do with [ef41cdc](https://github.com/taxprofiler/taxpasta/pull/31/commits/ef41cdcd80dacc658a9a829443009561d5264de0)

Reply:
That's the default anyway but fine to be explicit about it.